website_sale_customer_type/views/auth_signup_login_templates.xml

```diff
@@ -4,20 +4,27 @@
 <odoo>
     <template id="customer_type_fields" inherit_id="auth_signup.fields">
         <xpath expr="//div[hasclass('field-login')]" position="before">
-            <div class="form-group field-customer-type">
-                <label for="customer_type" class="control-label">Your Customer Type</label>
-                <input type="text"
-                       id="customer_type"
-                       name="customer_type"
-                       class="form-control"
-                       t-att-value="customer_type.name if customer_type else None"
-                       autocapitalize="off"
-                       required="required"
-                       readonly="'readonly'"/>
-            </div>
+            <t t-if="is_signup">
+                <div class="form-group field-customer-type">
+                    <label for="customer_type" class="control-label">Your Customer Type</label>
+                    <input type="text"
+                           id="customer_type"
+                           name="customer_type"
+                           class="form-control"
+                           t-att-value="customer_type.name if customer_type else None"
+                           autocapitalize="off"
+                           required="required"
+                           readonly="readonly"/>
+                    <input type="hidden"
+                           id="customer_type_id"
+                           name="customer_type_id"
+                           t-att-value="customer_type.id if customer_type else 0"
+                           readonly="readonly"/>
```

Comment:
@vvrossem Here, check how I use `hidden` input to get id of a `customer_type` and keep showing the name of the `customer_type` to the user. :)